backend/src/main/java/gr/uom/strategicplanning/controllers/OrganizationController.java

```diff
@@ -510,19 +510,13 @@ public ResponseEntity<Map> getCodeSmellsDistribution(@PathVariable Long id) {
             OrganizationAnalysis organizationAnalysis = organization.getOrganizationAnalysis();
             TechDebtStats techDebtStats = organizationAnalysis.getTechDebtStats();
 
-            Collection<OrganizationCodeSmellDistribution> codeSmellsDistribution = techDebtStats.getCodeSmells();
-
-            int totalCodeSmells = 0;
-
-            for (OrganizationCodeSmellDistribution codeSmell : codeSmellsDistribution) {
-                totalCodeSmells += codeSmell.getCount();
-            }
-
-
             Map<String, Object> codeSmellsDistributionResponse = new HashMap<>();
+
+            int totalCodeSmells = techDebtStats.getTotalCodeSmells();
             codeSmellsDistributionResponse.put("totalCodeSmells", totalCodeSmells);
-            codeSmellsDistributionResponse.put("codeSmellsDistribution", codeSmellsDistribution);
 
+            Map<String, Integer> codeSmellsDistribution = techDebtStats.getCodeSmells();
```

Comment:
In order to not change the response structure, do this instead:

            Map<String, Integer> codeSmellsDistributionMap = techDebtStats.getCodeSmells();
            List<OrganizationCodeSmellDistribution> codeSmellsDistribution = new ArrayList<>();
            for (Map.Entry<String, Integer> entry : codeSmellsDistributionMap.entrySet()) {
                codeSmellsDistribution.add(new OrganizationCodeSmellDistribution(entry.getKey(),entry.getValue()));
            }


Where OrganizationCodeSmellDistribution can be a DTO class:

            public class OrganizationCodeSmellDistribution {
                private String severity;
                private int count;
            }